go/runtime/rofl/api/attestation.go

```diff
@@ -0,0 +1,51 @@
+package api
+
+import (
+	"github.com/oasisprotocol/oasis-core/go/common/cbor"
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
+)
+
+const (
+	// LocalRPCEndpointAttestation is the name of the local RPC endpoint for attestation functions.
+	LocalRPCEndpointAttestation = "attestation"
+
+	// MethodAttestLabels is the name of the AttestLabels method.
+	MethodAttestLabels = "AttestLabels"
+
+	// MaxAttestLabels is the maximum number of labels on can request attestation for.
+	MaxAttestLabels = 10
+)
+
+// AttestLabelsRequest is a request to host to attest to specific component labels.
+type AttestLabelsRequest struct {
+	// Labels are the labels to attest to.
+	Labels []string `json:"labels"`
+}
+
+// AttestLabelsResponse is the response from the AttestLabels method.
+type AttestLabelsResponse struct {
+	// Labels are the attested labels, in order.
+	Labels []*AttestedLabel `json:"labels"`
+	// NodeID is the public key of the node attesting to the labels.
+	NodeID signature.PublicKey `json:"node_id"`
+	// Signature is the signature of the attested labels.
+	Signature signature.RawSignature `json:"signature"`
+}
+
+// AttestLabelsSignatureContext is the signature context used for label attestation.
+var AttestLabelsSignatureContext = signature.NewContext("oasis-core/node: attest component labels")
+
+// AttestedLabel is a label key/value pair that is used for label attestation.
+type AttestedLabel struct {
+	_ struct{} `cbor:",toarray"` //nolint
+
+	// Key is the label key.
+	Key string `json:"key"`
+	// Value is the label value.
+	Value string `json:"value"`
+}
+
+// AttestLabels signs the given list of labels and returns the signature.
+func AttestLabels(signer signature.Signer, labels []*AttestedLabel) (*signature.RawSignature, error) {
+	return signature.SignRaw(signer, AttestLabelsSignatureContext, cbor.Marshal(labels))
```

Comment:
If two components have the same labels, the same attest response can be used. Should we include component ID in the signature?

Reply:
Ah yes, we need to include the component RAK.